engine/src/render/shader/shader.cpp

```diff
@@ -19,3 +19,8 @@ cubos::core::gl::ShaderStage Shader::shaderStage() const
 {
     return mShaderStage;
 }
+
+const ShaderBuilder& Shader::builder() const
+{
+    return mBuilder;
+}
```

Comment:
```suggestion
ShaderBuilder Shader::builder() const
{
    return mBuilder; // this or maybe call the constructor directly an avoid storing builder as member variable
}
```
I think maybe would be better to make the functions in ShaderBuilder mutate the class and here return a new instance of the ShaderBuilder

Reply:
Makes sense, indeed after some quick research I realized builders following this kind of pattern are usually mutable.